blocks/graph.py

```diff
@@ -204,10 +204,15 @@ def replace(self, replacements):
         # Sort `replacements` in topological order
         for node in apply_nodes_sorted:
             for input_ in node.inputs:
-                if input_ not in replacements:
-                    continue
-                replacement_keys_cur.append(input_)
-                replacement_vals_cur.append(replacements[input_])
+                if input_ in replacements:
```

Comment:
Sure, I will. I got it when if was monitoring mean activation of a dropped-out layer, that is the values from this layers were used for monitoring and cost computations.

It took me a while to find this bug and this is the main reason why I am lagging behind in a few important discussions here.